share/CMakeJS.cmake

```diff
@@ -0,0 +1,640 @@
+            message(SEND_ERROR "Cannot add file '${input}': File must be in a subdirectory of ${ARG_BASE_DIRS}")
+            return()
+        endif()
+
+        set(rel_file "${ARG_BASE_DIRS}/${relpath}")
+        _cmakejs_normalize_path(rel_file)
+        get_filename_component(source_file "${input}" ABSOLUTE)
+        # If we are here, source file is valid. Add IDE support
+        source_group("${name}" FILES "${source_file}")
+
+        if(DEFINED ARG_INTERFACE)
+            foreach(item IN LISTS ARG_INTERFACE)
+                target_sources(${name} INTERFACE "${source_file}")
+            endforeach()
+        endif()
+
+        if(DEFINED ARG_PRIVATE)
+            foreach(item IN LISTS ARG_PRIVATE)
+                target_sources(${name} PRIVATE "${source_file}")
+            endforeach()
+        endif()
+
+        if(DEFINED ARG_PUBLIC)
+            foreach(item IN LISTS ARG_PUBLIC)
+                target_sources(${name} PUBLIC "${source_file}")
+            endforeach()
+        endif()
+
+        foreach(input IN LISTS ARG_UNPARSED_ARGUMENTS)
+            target_sources(${name} PRIVATE "${source_file}")
+        endforeach()
+
+    endforeach()
+
+endfunction()
+
+#[=============================================================================[
+Add pre-processor definitions to an existing Napi Addon target.
+
+cmakejs_napi_addon_add_definitions(<name> [items1...])
+cmakejs_napi_addon_add_definitions(<name> <INTERFACE|PUBLIC|PRIVATE> [items1...] [<INTERFACE|PUBLIC|PRIVATE> [items2...] ...])
+]=============================================================================]#
+function(cmakejs_napi_addon_add_definitions name)
+
+    # Check that this is a Node Addon target
+    get_target_property(is_addon_lib ${name} ${name}_IS_NAPI_ADDON_LIBRARY)
+    if(NOT TARGET ${name} OR NOT is_addon_lib)
+        message(SEND_ERROR "'cmakejs_napi_addon_add_definitions()' called on '${name}' which is not an existing napi addon library")
+        return()
+    endif()
+
+    set(options)
+    set(args)
+    set(list_args INTERFACE PRIVATE PUBLIC)
+    cmake_parse_arguments(ARG "${options}" "${args}" "${list_args}" "${ARGN}")
+
+    if(DEFINED ARG_INTERFACE)
+        foreach(item IN LISTS ARG_INTERFACE)
+            target_compile_definitions(${name} INTERFACE "${item}")
+        endforeach()
+    endif()
+
+    if(DEFINED ARG_PRIVATE)
+        foreach(item IN LISTS ARG_PRIVATE)
+            target_compile_definitions(${name} PRIVATE "${item}")
+        endforeach()
+    endif()
+
+    if(DEFINED ARG_PUBLIC)
+        foreach(item IN LISTS ARG_PUBLIC)
+            target_compile_definitions(${name} PUBLIC "${item}")
+        endforeach()
+    endif()
+
+    foreach(input IN LISTS ARG_UNPARSED_ARGUMENTS)
+        target_compile_definitions(${name} "${item}")
+    endforeach()
+
+endfunction()
+
+set (CMAKEJS_TARGETS)
+list (APPEND CMAKEJS_TARGETS
+  node-dev
+  node-api
+  node-addon-api
+  cmake-js
+)
+export (
```

Comment:
I dont have any idea what this and below is doing right now. Not things I have encountered before

Reply:
Check the build directory 'share' dir. 

It's all in the [official topic](https://cmake.org/cmake/help/book/mastering-cmake/cmake/Help/guide/importing-exporting/index.html).

That [guide](https://cmake.org/cmake/help/book/mastering-cmake/index.html) is very useful.